### ISSUE TITLE
[Bug] Ramdom dataset mismatched token length with conifg

### ISSUE THREAD

**Issue by JamesBrianD** [sgl-project/sglang-jax#272]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

When testing sglang-jax and vllm-tpu, it was observed that while the input length was set to 1024, the actual prompt length accepted by the server fluctuated and was not consistently 1024. This discrepancy arises because `seq != tokenizer.decode(tokenizer.encode(seq))`. For reference, see this PR https://github.com/vllm-project/vllm/pull/24937

### Reproduction

Client: `python3 -m sgl_jax.bench_serving --backend sgl-jax --dataset-name random --num-prompts 96 --random-input 1024 --random-output 1024 --random-range-ratio 1 --warmup-requests 0 --max-concurrency 32`. 
Server: The discrepancy between the prompt length from the print server and the input length of 1024 can be observed.

### Environment

```
Python: 3.12.12 (main, Oct 14 2025, 21:25:31) [Clang 20.1.4 ]
sglang-jax: 0.0.2
jax: 0.8.0
jaxlib: 0.8.0
triton: 3.5.0
transformers: 4.55.4
numpy: 2.2.6
aiohttp: 3.13.1
fastapi: 0.116.2
huggingface_hub: 0.34.6
modelscope: 1.28.2
orjson: 3.11.3
packaging: 25.0
psutil: 7.0.0
pydantic: 2.11.10
python-multipart: 0.0.20
pyzmq: 27.0.2
uvicorn: 0.35.0
uvloop: 0.21.0
openai: 2.6.0
tiktoken: 0.12.0
[TPU v6 lite-0]: TPU_0(process=0,(0,0,0,0))
[TPU v6 lite-1]: TPU_1(process=0,(1,0,0,0))
[TPU v6 lite-2]: TPU_2(process=0,(0,1,0,0))
[TPU v6 lite-3]: TPU_3(process=0,(1,1,0,0))
ulimit soft: 1048576
```

**Comment by JamesBrianD** [comment]
Fixed in https://github.com/sgl-project/sglang-jax/pull/274
